xrdb.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * xrdb.c - X Resources DataBase communication functions
+ *
+ * Copyright © 2015 Yauhen Kirylau <yawghen@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "xrdb.h"
+#include "globalconf.h"
+
+#include <X11/Xresource.h>
+
+#include <string.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+Display *dpy;
+XrmDatabase xrmdb;
+
+/* \brief open X display and X Resources DB
+ */
+static void xrdb_init(void) {
+  /**/
+  /* Open display with Xlib */
+  /**/
+  char *displayname = NULL;
```

Comment:
http://www.x.org/wiki/guide/xlib-and-xcb/#index2h2 says
> As mentioned above, XCB provides a method for incremental conversion from Xlib to XCB. One can use libX11 to open the display and pass the Display pointer it returns to existing code, toolkits, and libraries. To call an XCB function, one can convert the Display pointer to an xcb_connection_t pointer for the same connection. This enables calling into Xlib and XCB from the same application.

Not sure if that works also in the other direction?!